src/olympia/addons/tests/test_views.py

```diff
@@ -852,6 +852,47 @@ def test_too_many_categories(self):
         )
         assert response.status_code == 201, response.content
 
+    def test_set_slug(self):
+        response = self.client.post(
+            self.url,
+            data={
+                **self.minimal_data,
+                'slug': '!@!#!@##@$$%$#%#%$^^%&%',
+            },
+        )
+        assert response.status_code == 400, response.content
+        assert response.data == {
+            'slug': [
+                'Enter a valid “slug” consisting of letters, numbers, underscores or '
+                'hyphens.'
+            ]
+        }
+
+        DeniedSlug.objects.create(name='denied-slug')
+        response = self.client.post(
+            self.url,
+            data={
+                **self.minimal_data,
+                'slug': 'denied-slug',
+            },
+        )
+        assert response.status_code == 400, response.content
+        assert response.data == {
+            'slug': ['This slug cannot be used. Please choose another.']
+        }
+
+        response = self.client.post(
+            self.url,
+            data={
+                **self.minimal_data,
+                'slug': '1234',
+            },
+        )
+        assert response.status_code == 400, response.content
+        assert response.data == {
+            'slug': ['This slug cannot be used. Please choose another.']
+        }
+
```

Comment:
Could you add a test for the `addon with this slug already exists` case ? (I assume this is already handled by DRF because it's unique, but it's good to be sure)

Reply:
https://github.com/mozilla/addons-server/pull/18299/commits/f8f35438a6ffef4e8e5184f8ee67d4d44294f064